src/main/scala/io/iohk/ethereum/network/p2p/messages/CommonMessages.scala

```diff
@@ -0,0 +1,117 @@
+        case _ => throw new RuntimeException("Cannot decode Status")
+      }
+    }
+
+    val code: Int = 0x10 + 0x00
+  }
+
+  case class Status(protocolVersion: Int, networkId: Int, totalDifficulty: BigInt, bestHash: ByteString, genesisHash: ByteString) extends Message {
+    override def code: Int = Status.code
+
+    override def toString: String = {
+      s"""Status {
+         |protocolVersion: $protocolVersion
+         |networkId: $networkId
+         |totalDifficulty: $totalDifficulty
+         |bestHash: ${Hex.toHexString(bestHash.toArray[Byte])}
+         |genesisHash: ${Hex.toHexString(genesisHash.toArray[Byte])}
+         |}""".stripMargin
+    }
+  }
+
+  object Transactions {
+    implicit val rlpEndDec = new RLPEncoder[Transactions] with RLPDecoder[Transactions] {
+      override def encode(obj: Transactions): RLPEncodeable = {
+        import obj._
+        RLPList(txs.map(Transaction.rlpEndDec.encode): _*)
+      }
+
+      override def decode(rlp: RLPEncodeable): Transactions = rlp match {
+        case rlpList: RLPList => Transactions(rlpList.items.map(Transaction.rlpEndDec.decode))
+        case _ => throw new RuntimeException("Cannot decode Transactions")
+      }
+    }
+
+    val code: Int = 0x10 + 0x02
+  }
+
+  case class Transactions(txs: Seq[Transaction]) extends Message {
+    override def code: Int = Transactions.code
+  }
+
+  object Transaction {
+    implicit val rlpEndDec = new RLPEncoder[Transaction] with RLPDecoder[Transaction] {
+      override def encode(obj: Transaction): RLPEncodeable = {
+        import obj._
+        RLPList(nonce, gasPrice, gasLimit, receivingAddress.toArray[Byte], value,
+          payload.fold(_.byteString.toArray[Byte], _.byteString.toArray[Byte]),
+          pointSign, signatureRandom.toArray[Byte], signature.toArray[Byte])
+      }
+
+      override def decode(rlp: RLPEncodeable): Transaction = rlp match {
+        case RLPList(nonce, gasPrice, gasLimit, (receivingAddress: RLPValue), value, payload, pointSign, signatureRandom, signature)
+          if receivingAddress.bytes.nonEmpty =>
+          Transaction(nonce, gasPrice, gasLimit, ByteString(receivingAddress: Array[Byte]), value, Right(TransactionData(ByteString(payload: Array[Byte]))),
+            pointSign, ByteString(signatureRandom: Array[Byte]), ByteString(signature: Array[Byte]))
+
+        case RLPList(nonce, gasPrice, gasLimit, (receivingAddress: RLPValue), value, payload, pointSign, signatureRandom, signature)
+          if receivingAddress.bytes.isEmpty =>
+          Transaction(nonce, gasPrice, gasLimit, ByteString(), value, Left(ContractInit(ByteString(payload: Array[Byte]))),
+            pointSign, ByteString(signatureRandom: Array[Byte]), ByteString(signature: Array[Byte]))
+
+        case _ => throw new RuntimeException("Cannot decode Transaction")
+      }
+    }
+  }
+
+  //ETH yellow paper section 4.3
+  case class Transaction(
```

Comment:
ok, I get it now
I think that this case class `Transaction` belongs to messages because it is used to decode messages about transactions received from other nodes